### DISCUSSION TITLE
[HxGrid] dynamically show/hide columns with checkbox menu

### DISCUSSION THREAD

**Discussion by econner20** [havit/Havit.Blazor#781] [Q&A]
Please see attached screenshot. Is it possible to duplicate this feature with HxGrid? Are there any examples or documentation on how to add a icon to a column header? Also, Is it possible to use CheckboxList with the icon to show/hide columns by setting the column visible property based on the checked field in the drop down checkbox menu..

![FieldShowHide](https://github.com/havit/Havit.Blazor/assets/3165802/eefc9f6c-615b-4063-97dd-203017c3fa44)

**Comment by hakenr** [comment]
Absolutely, this is entirely feasible. You simply need to design the UI with a list of columns and then apply the settings collected there to the `<Columns>` parameter of the `HxGrid`. Here's an example from one of our projects (UI in Czech language) that includes the ability to change the order of columns:

![image](https://github.com/havit/Havit.Blazor/assets/12828888/b8d5375f-5fb2-4429-8b17-740d1a33e175)


Unfortunately, providing a full demo/sample exceeds our current capabilities without compensation. If you require further assistance, we're available for commercial support. Additionally, any support for our library through GitHub sponsorship is greatly appreciated and could enable us to prepare such demos.

**Reply by econner20** [reply]
Thanks. I think I found what I need with HeaderTemplate, Dropdown button and CheckboxList
